src/commands/update.js

```diff
@@ -55,4 +60,29 @@ export default class Update extends Command {
       await logChopper.chop(this.out.errlog)
     } catch (e) { this.out.debug(e.message) }
   }
+
+  async generateAutocompleteCommands () {
+    try {
+      // TODO: move from cli to client dir if not already present
+      // if (!fs.pathExistsSync(path.join(this.config.dataDir, 'client', 'autocomplete', 'bash', 'heroku'))) {
+      //   const cli = path.join(this.config.dataDir, 'cli', 'autocomplete')
+      //   const client = path.join(this.config.dataDir, 'client', 'autocomplete')
+      //   fs.copySync(cli, client)
+      // }
+      const plugins = await new Plugins(this.out).list()
+      const cmds = plugins.map(p => p.commands.map(c => {
+        if (c.hidden) return
+        let plublicFlags = Object.keys(c.flags).filter(flag => !c.flags[flag].hidden).map(flag => `--${flag}`).join(' ')
+        let flags = plublicFlags.length ? ` ${plublicFlags}` : ''
+        let namespace = p.namespace ? `${p.namespace}:` : ''
+        return `${namespace}${c.id}${flags}`
+      }))
+      const commands = flatten(cmds).filter(c => !!c).join('\n')
+      fs.mkdirpSync(path.join(this.config.dataDir, 'client', 'autocomplete'))
```

Comment:
prefer `await fs.mkdirp()` and `await fs.writeFile()`

good to get in the habit of using async when we can